pvlib/modelchain.py

```diff
@@ -896,33 +998,45 @@ def infer_temperature_model(self):
         elif {'noct_installed'} <= params:
             return self.fuentes_temp
         else:
-            raise ValueError('could not infer temperature model from '
-                             'system.temperature_module_parameters {}.'
-                             .format(self.system.temperature_model_parameters))
+            raise ValueError(f'could not infer temperature model from '
+                             f'system.temperature_model_parameters. Check '
+                             f'that all Arrays in system.arrays have '
+                             f'parameters for the same model (or models) '
```

Comment:
Probably better to keep this focused on temperature model params, so end with only "for the same temperature model. Common temperature model parameters: {params}".